File: languageExtensions/OrchidSyntaxHighlighter/src/main/java/com/eden/orchid/languages/highlighter/SyntaxHighlighterModule.java
package com.eden.orchid.languages.highlighter;

import com.eden.orchid.api.registration.OrchidModule;
import com.eden.orchid.api.resources.resourceSource.PluginResourceSource;
import com.eden.orchid.api.theme.components.OrchidComponent;
import org.jtwig.functions.JtwigFunction;

public class SyntaxHighlighterModule extends OrchidModule {

    @Override
    protected void configure() {
        addToSet(PluginResourceSource.class,
                HighlightResourceSource.class);

        // Syntax highlighting at built-time via Pygments
        addToSet(JtwigFunction.class,
                HighlightFilter.class,
                HighlightPrismFilter.class);

        // Syntax Highlighting at runtime via Prism
        addToSet(OrchidComponent.class,
                PrismComponent.class);
    }
}

File: languageExtensions/OrchidSyntaxHighlighter/src/main/kotlin/com/eden/orchid/languages/highlighter/SyntaxHighlighterModule.kt
package com.eden.orchid.languages.highlighter

import com.eden.orchid.api.compilers.TemplateTag
import com.eden.orchid.api.registration.OrchidModule
import com.eden.orchid.api.theme.components.OrchidComponent
import com.eden.orchid.languages.highlighter.components.PrismComponent
import com.eden.orchid.languages.highlighter.tags.HighlightTag
import com.eden.orchid.utilities.addToSet

class SyntaxHighlighterModule : OrchidModule() {

    override fun configure() {
        withResources(750)

        addToSet<TemplateTag, HighlightTag>()
        addToSet<OrchidComponent, PrismComponent>()
    }
}
